### PR TITLE
[Qwen3-Next][Perf][Attention] Replace torch.zeros with torch.empty to reduce overhead

### PR DESCRIPTION
**Pull request by ZJY0516**
## Purpose
As https://github.com/vllm-project/vllm/pull/26680 and https://github.com/vllm-project/vllm/pull/19784 mentioned, using `torch.zeros` to allocate attention output buffer will introduce an unnecessary kernel overhead.

> w/ torch.empty, we allocate but do not initialize. CUDAGraph would remove this allocation time.
w/ torch.zero, we allocate AND initialize, which requires an extra cuda/triton kernel. CUDAGraph cannot remove this kernel. This kernel adds ~1 us latency, which is on-par with a layer norm or rope kernel (~1.6 us).

qwen3-next as an example:
main
image
this pr
image

A triton kernel was eliminated

## accuracy test

```
vllm serve Qwen/Qwen3-Next-80B-A3B-Instruct --enable-expert-parallel -tp 4
lm_eval --model local-completions --model_args model=Qwen/Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8000/v1/completions -t gsm8k --num_fewshot 5 --batch_size 250
```

Tasks | Version | Filter | n-shot | Metric |   | Value |   | Stderr
-- | -- | -- | -- | -- | -- | -- | -- | --
gsm8k | 3 | flexible-extract | 5 | exact_match | ↑ | 0 | ± | 0
  |   | strict-match | 5 | exact_match | ↑ | 0 | ± | 0

### qwen3-next

TL;DR: throughput 6340.16 -> 6756.52

```
vllm serve Qwen/Qwen3-Next-80B-A3B-Instruct --enable-expert-parallel -tp 4
vllm bench serve \
--model Qwen/Qwen3-Next-80B-A3B-Instruct \
--dataset-name random \
--num-prompts 32 \
--random-input-len 2048 \
--random-output-len 1024
```
this pr
```
============ Serving Benchmark Result ============
Successful requests:                     32        
Failed requests:                         0         
Benchmark duration (s):                  14.55     
Total input tokens:                      65536     
Total generated tokens:                  32768     
Request throughput (req/s):              2.20      
Output token throughput (tok/s):         2252.17   
Peak output token throughput (tok/s):    2912.00   
Peak concurrent requests:                32.00     
Total Token throughput (tok/s):          6756.52   
---------------Time to First Token----------------
Mean TTFT (ms):                          1819.22   
Median TTFT (ms):                        1847.39   
P99 TTFT (ms):                           2960.06   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          12.39     
Median TPOT (ms):                        12.38     
P99 TPOT (ms):                           13.63     
---------------Inter-token Latency----------------
Mean ITL (ms):                           12.40     
Median ITL (ms):                         11.33     
P99 ITL (ms):                            12.00     
==================================================
```

main
```
============ Serving Benchmark Result ============
Successful requests:                     32        
Failed requests:                         0         
Benchmark duration (s):                  15.14     
Total input tokens:                      65536     
Total generated tokens:                  30472     
Request throughput (req/s):              2.11      
Output token throughput (tok/s):         2012.31   
Peak output token throughput (tok/s):    2400.00   
Peak concurrent requests:                32.00     
Total Token throughput (tok/s):          6340.16   
---------------Time to First Token----------------
Mean TTFT (ms):                          1355.49   
Median TTFT (ms):                        1365.46   
P99 TTFT (ms):                           2122.49   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          13.42     
Median TPOT (ms):                        13.32     
P99 TPOT (ms):                           14.34     
---------------Inter-token Latency----------------
Mean ITL (ms):                           13.46     
Median ITL (ms):                         12.74     
P99 ITL (ms):                            13.36     
==================================================
```

<details>
<summary> kimi linear has performance regression </summary>
### kimi linear

```
vllm serve moonshotai/Kimi-Linear-48B-A3B-Instruct --trust-remote-code -tp 4 --enable-expert-parallel
vllm bench serve \
--model moonshotai/Kimi-Linear-48B-A3B-Instruct \
--dataset-name random \
--num-prompts 32 \
--random-input-len 2048 \
--random-output-len 1024 \
--trust-remote-code
```
this pr
```
============ Serving Benchmark Result ============
Successful requests:                     32        
Failed requests:                         0         
Benchmark duration (s):                  11.18     
Total input tokens:                      65536     
Total generated tokens:                  31784     
Request throughput (req/s):              2.86      
Output token throughput (tok/s):         2843.43   
Peak output token throughput (tok/s):    3424.00   
Peak concurrent requests:                32.00     
Total Token throughput (tok/s):          8706.36   
---------------Time to First Token----------------
Mean TTFT (ms):                          1028.07   
Median TTFT (ms):                        1031.84   
P99 TTFT (ms):                           1634.19   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          9.95      
Median TPOT (ms):                        9.89      
P99 TPOT (ms):                           11.42     
---------------Inter-token Latency----------------
Mean ITL (ms):                           9.91      
Median ITL (ms):                         9.32      
P99 ITL (ms):                            10.10     
==================================================
```

main
```
============ Serving Benchmark Result ============
Successful requests:                     32        
Failed requests:                         0         
Benchmark duration (s):                  11.06     
Total input tokens:                      65536     
Total generated tokens:                  32206     
Request throughput (req/s):              2.89      
Output token throughput (tok/s):         2911.00   
Peak output token throughput (tok/s):    3552.00   
Peak concurrent requests:                32.00     
Total Token throughput (tok/s):          8834.59   
---------------Time to First Token----------------
Mean TTFT (ms):                          1179.73   
Median TTFT (ms):                        1196.33   
P99 TTFT (ms):                           1800.05   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          9.65      
Median TPOT (ms):                        9.62      
P99 TPOT (ms):                           10.65     
---------------Inter-token Latency----------------
Mean ITL (ms):                           9.64      
Median ITL (ms):                         9.05      
P99 ITL (ms):                            9.59      
==================================================
```
```
vllm serve moonshotai/Kimi-Linear-48B-A3B-Instruct --trust-remote-code -tp 4
lm_eval --model local-completions --model_args model=moonshotai/Kimi-Linear-48B-A3B-Instruct,base_url=http://localhost:8000/v1/completions,trust_remote_code=True -t gsm8k --num_fewshot 5 --batch_size 250
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8939|±  |0.0085|
|     |       |strict-match    |     5|exact_match|↑  |0.8764|±  |0.0091|

</details>

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by ZJY0516** [comment]
I don't know why we have performance regression on kimi linear. Do you have any idea? @BoyuanFeng

**Comment by BoyuanFeng** [comment]
image

Looks like there are still 3 kernels?

For the perf, would the number be stable if you run it twice?

**Comment by ZJY0516** [comment]
> Looks like there are still 3 kernels?

I think `triton_poi_fused_5` is generated by `torch.zeros`?

**Comment by vadiklyutiy** [comment]
@codex review

**Comment by vadiklyutiy** [comment]
Could you try 
```
lm_eval --model local-chat-completions --model_args model=Qwen/Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8000/v1/chat/completions,num_concurrent=250 --tasks gsm8k --apply_chat_template --num_fewshot 5
```

and maybe different values of `num_concurrent`. It should produce more variation of shapes in `forward` step.

**Comment by ZJY0516** [comment]
I ran tests on the main branch and observed inconsistent results across different commands.

```
lm_eval --model local-completions --model_args model=Qwen/Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8000/v1/completions -t gsm8k --num_fewshot 5 --batch_size 250
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8537|±  |0.0097|
|     |       |strict-match    |     5|exact_match|↑  |0.8074|±  |0.0109|

```
lm_eval --model local-chat-completions --model_args model=Qwen/Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8000/v1/chat/completions,num_concurrent=280 --tasks gsm8k --apply_chat_template --num_fewshot 5
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.7877|±  |0.0113|
|     |       |strict-match    |     5|exact_match|↑  |0.6566|±  |0.0131|
```
lm_eval --model local-completions --model_args model=Qwen/Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8000/v1/completions -t gsm8k --num_fewshot 5 --batch_size 250 --apply_chat_template
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.7998|±  | 0.011|
|     |       |strict-match    |     5|exact_match|↑  |0.6664|±  | 0.013|

**Comment by QilaiZhang** [comment]
@ZJY0516 Hi! I've tested this PR on the latest main branch and didn't observe any accuracy issues. I'm not sure why our test results differ. Looking forward to your reply, thanks!
```
vllm serve Qwen/Qwen3-Next-80B-A3B-Instruct --enable-expert-parallel -tp 4
lm_eval --model local-chat-completions --model_args model=Qwen3-Next-80B-A3B-Instruct,base_url=http://localhost:8898/v1/chat/completions,num_concurrent=250 --tasks gsm8k --apply_chat_template --num_fewshot 5
```
### Main
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8423|±  |0.0100|
|     |       |strict-match    |     5|exact_match|↑  |0.6384|±  |0.0132|

### This PR

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8446|±  |0.0100|
|     |       |strict-match    |     5|exact_match|↑  |0.6429|±  |0.0132|
